client/my-sites/github-deployments/deployments/deployment-status.tsx

```diff
@@ -36,10 +38,15 @@ function getText( status: DeploymentStatusValue ) {
 	}
 }
 
-export const DeploymentStatus = ( { status }: DeploymentStatusProps ) => {
+export const DeploymentStatus = ( { status, onStatusClick }: DeploymentStatusProps ) => {
 	return (
-		<div className={ `github-deployments-status github-deployments-status__${ status }` }>
+		<Button
+			plain
+			compact
+			className={ `github-deployments-status github-deployments-status__${ status }` }
+			onClick={ onStatusClick }
```

Comment:
This could be a link instead of a button event handler. This way the user could open it in a different tab for example.